### PR TITLE
add community governance rules and membership list

### LINKED ISSUE #167: Community Governance Proposal
# 前言
Layotto刚刚发布了第一个版本，社区各位同学做出了很多贡献。为了回报大家的努力，最好的方式就是把社区越做越大（毕竟大家都希望自己写的代码对世界产生更大的影响力，而不仅仅是一两个公司的自high项目)

那么怎么才能把社区做大呢？其中一件能做的事情就是制定好的社区规则。个人理解，好的社区规则要足够透明、开放，允许每个感兴趣的人加入、提建议、参与决策；好的社区规则要让大家能看到投入后的回报，回报有很多种，例如学到感兴趣的知识，实践感兴趣的技术，获得更酷的荣誉 title，获得社区影响力，自己写的代码跑在很多公司的机器上从而获得成就感，成为社区的决策者，获得求职时的亮眼项目履历，参与社区活动获得奖品IDE License……

Layotto 是 MOSN 社区的项目，而 SOFAStack 社区和 MOSN 社区之前用的都是同一套组织架构（如果我理解的不对请指正哈），针对过去的一些问题，本草案试着对规则提出一些变化，希望调动大家积极性，让项目 owner 觉得是在为个人爱好/个人荣誉而参与社区，让每个感兴趣的技术同学都能参与到社区核心决策中，学到自己感兴趣的知识、做自己感兴趣的项目，成为项目的决策者之一，and finally impact the world!

# 目前 SOFAStack 组织架构
[https://www.sofastack.tech/community/](https://www.sofastack.tech/community/)

存在的问题

1. 虽有 User Group 但活跃度不足
1. Contributor 晋升路线过长、模糊
1. 项目 owner 投入开源的时间有限（加班多，也没办法……）、动力不足（最直接的表现是，离职或转岗后缺少继续贡献社区的动力）

# v2.0 草案
image

新增角色用红色标识。
解释：
## 1. Title 社区化：所有社区 title、权限和参与者所属公司无关
此条为了调动大家积极性，让项目 owner 觉得是为了个人爱好/个人荣誉而参与社区，不管去哪家公司都还是SOFAStack PMC member/Co-founder; 让每个感兴趣的技术同学都能参与到社区核心决策中，促进社区活跃
### Co-founder（项目创始人）终身荣誉制
作为终身荣誉，但没有实权。比如某位开发者创建了项目A，多年后即使他不再参与社区维护、辗转跳槽多家公司，经历了风风雨雨、起起落落，决定转行不做程序员了、开了家火锅店，他也依然是项目A的 Co-founder

### PMC(Project Management Committee) member
PMC member 作为项目核心管理团队，有项目的决策权，**和所属公司无关，仅看社区活跃度**。比如，某 Co-founder 如果多年不参与社区维护，就不再是PMC member了。比如某位编程爱好者主业是火锅店老板、不是程序员，只要他对项目的贡献足够，他也可以是PMC member。至于如何成为 PMC member，见下面的晋升规则

### 欢迎各位项目 owner 给自己和其他人发 title
只有让项目 owner 和社区贡献者有动力持续维护，社区才会越做越大，因此欢迎大家给自己和其他人发 title，不需要谦虚。

​

### Active Contributor 和 Reviewer
Active Contributor 是一年贡献超过xxx 个 PR 的 Contributor。（每个项目自己定）
Reviewer 从 Active Contributor 中诞生，具有 Review PR 的义务，并且对某个项目/模块的 PR 的点赞（LGTM）有效。
​

### Core Team 核心开发者团队

SOFAStack & MOSN 社区的每个项目可以拉个独立的核心团队群(钉钉/微信都行)。
Core Team 负责某个项目的开发和维护工作，对代码的质量负责。
我们将邀请 Active Contributor 加入Core Team，开发者们将在 Core Team 中获得其他开发者的持续帮助、交流，一边锻炼技术能力，一边优化和完善项目。社区开发者们可通过 Core Team 逐渐从初始的 Active Contributor 成长为受到社区认可的 Reviewer、Committer 和 PMC member。
一般而言每个专项兴趣小组都会周期性的组织会议，讨论最近进展和遇到的问题，所有的会议讨论都公开在社区上，方便感兴趣的同学一起参与和讨论。
​

注：此项设计参考了Tidb社区的 [ Special Interest Group](https://pingcap.com/blog-cn/tidb-community-upgrade/#%E5%8F%98%E5%8C%96-3special-interest-group)

### 完善社区成员晋级的规则、指导机制
社区成员做出怎样的贡献后可以晋级？要有透明、具体的规则，让社区同学从 Contributor 成长到 Committer 或 Maintainer 有路可循，**让大家觉得“投入就一定有回报”**
具体晋升机制由每个项目的 Core Team 讨论决定

### 评审社区化：需求和技术方案公开评审
重要需求的评审、技术方案的评审应该在社区内**以开放的方式做评审，而不是蚂蚁员工开内部评审会**
例如要做技术方案评审时，社区成员可以在自己的用户群或者 Core Team 群里组织线上会议、开直播，允许每一个感兴趣的人观看、参与。如果觉得直播太重，issue 区发 proposal 也行。
​

否则，试想：某项目 owner 离开蚂蚁后，虽然他是 PMC member，但是方案评审都不带他，他还会有动力继续参与社区么？
​

### 晋升社区化
晋升由社区成员在群内组织投票，具体晋升规则由每个项目的核心团队自行协商决定
​

### 其他社区规则
例如冲突解决规则，code review 要几票才能通过等，由每个项目的核心团队自行协商决定
# 参考资料
Apollo 
[https://github.com/ctripcorp/apollo/pull/3670](https://github.com/ctripcorp/apollo/pull/3670)
​[https://github.com/ctripcorp/apollo/issues/3684](https://github.com/ctripcorp/apollo/issues/3684)
[https://github.com/ctripcorp/apollo/discussions/categories/announcements](https://github.com/ctripcorp/apollo/discussions/categories/announcements)
​

Tidb的社区组织架构 [https://pingcap.com/blog-cn/tidb-community-upgrade/](https://pingcap.com/blog-cn/tidb-community-upgrade/)

# 怎么落地？
如果大家讨论觉得没问题了，可以先从 Layotto 和 SOFA-registry 开始试运行该制度。
一开始，先由项目 owner 根据社区贡献程度来确定初始团队(给大家发 title)，再由初始团队讨论细化晋升规则、title 提名。
​

# 这是草案、不是结论，欢迎提出不同意见
本文仅作抛砖引玉、试着发出来看看大家意见，欢迎提出您的不同意见~
本人经验不足，上次正儿八经写这种组织规则还是在小学三年级玩网游《传奇》时创建公会，所以如有不妥快来指正
​

### PR DESCRIPTION
**Pull request by seeflood**
**What this PR does**:
Hi guys, according to #167 ,I submit this PR to announce our community roles,promotion rules and initial PMC member list
Welcome to give your advice or different opinions!
Meanwhile,if you are interested to join in as special roles like Member or Reviewer, please contact me.

Details:
Add community roles description. see `docs/zh/community/governance.md`
Add Layotto community promotion rules.see `docs/zh/community/promote.md`
Add Layotto community membership list. see `docs/zh/community/people.md`

**Which issue(s) this PR fixes**:

Fixes #167

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/zh/community/governance.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+#### Member分配的github组织权限
+默认权限，没有写权限。但是LGTM有效
+#### 成为Member的条件（加入组织的条件）
+每个项目自己定
+### 2.2. Reviewer
+#### 角色说明
+和Member一样，也是负责code review以及处理issue，算是对Member贡献的认可和感谢。长期review贡献多可以晋升Committer
+
+Reviewer要承担一定的review责任。比如某个issue/Pull request涉及某个模块，项目维护者可以request这个模块的Reviewer进行Review、负责把关。比如有人给Layotto的WASM模块提bug，可以request WASN模块的reviewer @zhenjunMa
```
Comment (tianjipeng) [inline docs/zh/community/governance.md:73]: > 可以request WASN模块的reviewer

WASN -> WASM

File: docs/zh/community/governance.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+a. 成为Reviewer之后，职责相比member要更明确，就是处理issue和PR，而且必须保持每周至少处理2个issue或1个PR，如果一个月内，有两周处理的频率没有达到标准，会回退到Member；
+
+b. 认为给Reviewer定目标太有KPI味了，工作背KPI就够难了别给社区搞KPI
+#### Reviewer分配的github组织权限
+![](https://user-images.githubusercontent.com/26001097/129857585-2f2ddcda-4a5d-4f94-a36d-48e6a9f52e0e.png)
+
+#### 添加这个角色的目的?
+1. 希望能让MOSN和SOFAStack社区响应速度更快一些，响应pr快一些（至于如何提高issue响应速度，我还没想到啥办法，欢迎提建议
+
```
Comment (zhenjunMa) [inline docs/zh/community/governance.md:88]: 空行

File: docs/zh/community/governance.md (hunk 1 above)
Comment (seeflood) [reply docs/zh/community/governance.md]: fixed

File: docs/zh/community/governance.md (hunk 2 above)
Comment (seeflood) [reply docs/zh/community/governance.md:88]: fixed

**Review by zhenjunMa — DISMISSED** [review]

**Comment by seeflood** [comment]
I added some options in Reviewer promotion requirement:
image

**Comment by seeflood** [comment]
I added Reviewer responsibilities. FYI @zhenjunMa @stulzq @ZLBer @MoonShining
image

**Comment by seeflood** [comment]
I added explanation for why it is designed this way.Welcome to give your advice
image

**Review by zhenjunMa — APPROVED** [review]

File: docs/zh/community/people.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+| RPC相关  | @MoonShining |
+
+
+## Member
+邀请中，欢迎感兴趣的同学联系@seeflood 加入社区
+![img.png](../../img/community/img.png)
+
+以及：
+
+![img.png](img.png)
```
Comment (nobodyiam) [inline docs/zh/community/people.md:25]: @seeflood @zhenjunMa 

Looks like this link is wrong, it won't display.

image

Comment (seeflood) [reply docs/zh/community/people.md:25]: Ok, it will be fixed in the #234 PR

**Pull request merged**